src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,76 @@
+package duke;
+
+/**
+ * Handles individual inputs and interprets the commands from the user.
+ */
+public class Parser {
+    public Parser() {
+    }
+
+    /**
+     * Only constructor for parser. Catches cases where the user inputs an invalid command.
+     *
+     * @param duke
+     * @param taskList
+     * @param ui
+     * @param storage
+     */
+    protected String activate(String input, Duke duke, TaskList taskList, Ui ui, Storage storage) {
+        try {
+            return handleCommand(duke, input, taskList, ui, storage);
+        } catch (Exception e) {
+            return String.format("%s\n☹ %s\n%s", Duke.line, e.getMessage(), Duke.line);
+        }
+    }
+
+    /**
+     * Listens to input. Parses the input into the command, body and date/time.
+     * Activates taskList to manage the task.
+     * Activates UI to respond to the user.
+     * Hanldes commands:
+     * "list", "save", "bye", "delete", "done", "todo" , "deadline", "event".
+     *
+     * @param duke
+     * @param currLine the current input from the user
+     * @param taskList
+     * @param ui
+     * @param storage
+     * @throws Exception when an invalid command is made. i.e. the first word in the input is
+     *                   invalid
+     * @return
+     */
+    protected String handleCommand(Duke duke, String currLine, TaskList taskList, Ui ui, Storage storage) throws Exception {
```

Comment:
Not sure if it's just me, but I feel that this method looks very cluttered. Also, missing return definition? Not sure if it's intentional (or if it's allowed).